### PR TITLE
Add support for htpdate

### PR DESCRIPTION
**Pull request by ricardosalveti**
Taking latest from meta-oe/master (due required changes), enable support for systemd and also provide a default config.

Systemd support should be sent upstream as well, will do that after finalizing this pr.

### CODE REVIEW

**Comment by ricardosalveti** [comment]
Not yet fully validated, will test it back on thursday (off this wednesday).

**Review by MrCry0 — APPROVED** [review]
LGTM

File: meta-lmp-base/recipes-support/htpdate/htpdate_1.3.6.bb [hunk 1]
```diff
@@ -28,4 +32,8 @@ do_compile () {
 
 do_install () {
 	oe_runmake install 'INSTALL=install' 'STRIP=echo' 'DESTDIR=${D}'
+	if ${@bb.utils.contains('DISTRO_FEATURES', 'systemd', 'true', 'false', d)}; then
+		install -d ${D}${systemd_unitdir}/system
+		install -m 644 ${S}/scripts/htpdate.service ${D}${systemd_unitdir}/system
```
Comment (quaresmajose) [inline meta-lmp-base/recipes-support/htpdate/htpdate_1.3.6.bb:41]: `${D}${systemd_system_unitdir}` is the same as `${D}${systemd_unitdir}/system` in a single variable.

File: meta-lmp-base/recipes-support/htpdate/htpdate/default.conf [hunk 2]
```diff
@@ -0,0 +1 @@
+HTPDATE_ARGS="-st www.example.com"
```
Comment (Tim-Anderson) [inline meta-lmp-base/recipes-support/htpdate/htpdate/default.conf:1]: Since this is a time update, shouldn't this be a real destination?

Comment (ricardosalveti) [reply meta-lmp-base/recipes-support/htpdate/htpdate/default.conf:1]: www.example.com is a valid destination, which is ok to be used as default, customers can change to any other address that might be valid for their own network.

**Review by angolini — APPROVED** [review]

File: meta-lmp-base/recipes-support/htpdate/htpdate_1.3.6.bb (hunk 1 above)
Comment (ricardosalveti) [reply meta-lmp-base/recipes-support/htpdate/htpdate_1.3.6.bb]: Done, I always forget about this one.

**Comment by ricardosalveti** [comment]
```
root@intel-corei7-64:~# htpdate -d -st www.example.com
www.example.com           80, 13 Oct 2022 22:52:41 GMT (126 ms) => 0
www.example.com           80, 13 Oct 2022 22:52:41 GMT (127 ms) => 0
www.example.com           80, 13 Oct 2022 22:52:41 GMT (127 ms) => 0
www.example.com           80, 13 Oct 2022 22:52:42 GMT (127 ms) => 1
when: 875000000, nap: -62500000
offset: 0.125000
Setting 0.125 seconds
Set time: Thu Oct 13 22:52:43 2022
root@intel-corei7-64:~# date
Thu Oct 13 22:52:45 UTC 2022
```

**Review by quaresmajose — APPROVED** [review]
LGTM

**Pull request merged**